### ISSUE TITLE
Creating a spatial index with invalid configuration leaves it in the database, rendering that index name unuseable.

### ISSUE THREAD

**Issue by jonathanwin** [neo4j-contrib/spatial#71]
Creating a spatial index with invalid configuration throws an IllegalArgumentException in LayerNodeIndex(), but the index configuration is still added to the IndexManager's internal store, with no way of removing it.
Removing the invalid index configuration requires deleting the whole database (via the published API at any rate)

Considering the IndexManager API, any operation on an index requires an Index object, therefore LayerNodeIndex should probably never throw an exception in it's constructor, but rather skip internal initialisation and declare itsself invalid somehow.

see https://groups.google.com/forum/#!topic/neo4j/SIxUsF2EExc "neo4j-spatial: Creating a WKT-based index for polygon geometries"

**Comment by craigtaverner** [comment]
Even though this issue was closed, the test case was not passing, it was just being ignored. I have now fixed this in a different way for the 0.15-neo4j-2.3 version. The exception we were throwing was never thrown at index creation time, only on first use, but we can instead use the fillInDefaults method to react at index creation. I made it that obvious and fixable issues in the config were fixed, and that unfixable issues threw exceptions early enough to prevent index creation from happening at all. Added two tests for these two scenarios.

See https://github.com/neo4j-contrib/spatial/commit/148c6232548ddb05d9efbb0d1e7ee864616484bb
